web/source/kmwembedded.ts

```diff
@@ -176,6 +176,42 @@ namespace com.keyman.osk {
   };
 }
 
+namespace com.keyman.text {
+  let nativeDefaultKeyOutput = Processor.prototype.defaultKeyOutput;
+
+  // Overrides the 'native'-mode implementation with in-app friendly defaults prioritized over 'native' defaults.
+  Processor.prototype.defaultKeyOutput = function(this: Processor, Lkc: KeyEvent, keyShiftState: number, usingOSK: boolean): string {
+    let keyman = com.keyman.singleton;
+
+    // Note:  this assumes Lelem is properly attached and has an element interface.
+    // Currently true in the Android and iOS apps.
+    let Codes = com.keyman.text.Codes;
+    let code = Lkc.Lcode;
+
+    // Default handling for external keys.
+    // Intentionally not assigning K_TAB or K_ENTER so KMW will pass them back
+    // to the mobile apps to handle (insert characters or navigate forms).
+    if(!usingOSK) { // If hardware-sourced.
+      if (code == Codes.keyCodes.K_SPACE) {
+        return ' ';
+      } else if (code == Codes.keyCodes.K_BKSP) {
+        keyman['interface'].defaultBackspace();
+        return '';
+      } else if (code == Codes.keyCodes.K_oE2) {
+        // Using defaults of English US layout for the 102nd key
+        if (Lkc.Lmodifiers == Codes.modifierCodes['SHIFT']) {
```

Comment:
I'll have to do more testing. In the emulator, the 102nd key was generating output.
But on my phone, the 102nd key didn't generate output.

Reply:
Ah, right, that part of code was called by both paths after all, not just the external keyboard path.  (Somehow I forgot that.)

Reply:
Yep, that's working correctly now